src/components/MobileFilter/index.tsx

```diff
@@ -20,14 +20,25 @@ export const MobileFilter: React.SFC<MobileFilterProps> = ({
     targetFilter
   }: CSEAdvancedFilterModels = filters;
 
+  const update = (selectedOptions: string[], code: FilterType) => {
+    if (selectedOptions.includes('All')) {
+      onUpdate([], code);
+    } else {
+      onUpdate(selectedOptions, code);
+    }
+  };
+
   const renderClaimFilter = (): JSX.Element | undefined => {
     let content: JSX.Element | undefined;
     if (claimFilter) {
+      let filterOptions: FilterOptionModel[] = [{ label: 'All', key: 'All', isSelected: false }];
+      filterOptions = filterOptions.concat(claimFilter.filterOptions);
+      claimFilter.filterOptions = filterOptions;
```

Comment:
This and the two lines above seem like a bit of unnecessary assignment. I would do something like
```js
const filterOptions: FilterOptionModel[] = [{ label: 'All', key: 'All', isSelected: false }];
claimFilter.filterOptions = filterOptions.concat(claimFilter.filterOptions);
```
Although I don't really like that `claimFilter` is being modified at all. It would be better to do something like

```js
const filterOptions: FilterOptionModely[] = [{ label: 'All', key: 'All', isSelected: false }];
const newFilter: AdvancedFilterCategoryModel = {
  filterOptions: filterOptions.concat(claimFilter.filterOptions),
  ...claimFilter
};
```
And then use `newFilter` to construct the `content` return value.